docs/chymyst07.md

```diff
@@ -728,4 +728,104 @@ In the chemical machine, input molecules to each reaction must be defined static
 Since `n` is a run-time parameter, we cannot generalize to `n` in this way.
 
 
+## Cigarette smokers
```

Comment:
Actually, I would suggest to move this documentation into another file - say `chymyst08.md` because I want to keep `chymyst07.md` for the smaller design patterns rather than larger examples like "dining philosophers" or "smokers".